File: objective-c/cpp_to_objc/SportsManagementObjC/SportsManagementObjC/Fighter.m
//
//  Fighter.m
//  SportsMgmtObjC
//
//  Created by Ming Chow on 1/26/12.
//  Copyright (c) 2012 Tufts University. All rights reserved.
//

#import "Fighter.h"

@implementation Fighter

@synthesize weight, affiliation, nickname;

-(id) initWith:(NSString *)fname and:(NSString *)lname and:(unsigned int)a and:(NSString *)e
{
    NSLog(@"Starting record for fighter...");
    self = [super initWithFirstName:fname withLastName:lname withAge:a withEmail:e];
    if (self) {
        weight = 0;
        affiliation = @"";
        nickname = @"";
    }
    return self;
}

-(void) print
{
    NSLog(@"First Name => %@", [self firstName]);
    NSLog(@"Last Name => %@", [self lastName]);
    NSLog(@"E-Mail Address => %@", [self email]);
    NSLog(@"Age => %i", [self age]);
    NSLog(@"Weight => %i", weight);
    NSLog(@"Nickname => %@", nickname);
    NSLog(@"Affiliation => %@", affiliation);
}

-(void) dealloc
{
    NSLog(@"...boo...");
    //[super release];
}

@end

File: examples/objective-c/cpp_to_objc/SportsManagement/SportsManagement/Fighter.cpp
//
//  Fighter.cpp
//

#include <iostream>
#include "Fighter.h"

using namespace std;

Fighter::Fighter (string firstName, string lastName, string email, unsigned int age):SportsClient(firstName, lastName, email, age)
{
    cout << "Starting record for fighter..." << endl;
    weight = 0;
    nickname = "";
    affiliation = "";
}

void Fighter::setWeight(unsigned int weight)
{
    this->weight = weight;
}

unsigned int Fighter::getWeight()
{
    return weight;
}

void Fighter::setNickname(string nickname)
{
    this->nickname = nickname;
}

string Fighter::getNickname()
{
    return nickname;
}

void Fighter::setAffiliation(string affiliation)
{
    this->affiliation = affiliation;
}

string Fighter::getAffiliation()
{
    return affiliation;
}

void Fighter::print()
{
    cout << "First Name => " << getFirstName() << endl;
    cout << "Last Name => " << getLastName() << endl;
    cout << "E-Mail Address => " << getEmail() << endl;
    cout << "Age => " << getAge() << endl;
    cout << "Weight => " << weight << endl;
    cout << "Nickname => " << nickname << endl;
    cout << "Affiliation => " << affiliation << endl;
}

Fighter::~Fighter()
{
    cout << "...boo..." << endl;
}
